x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,160 @@
+import { CLOUD_SECURITY_POSTURE_PACKAGE_NAME, STATUS_ROUTE_PATH } from '../../../common/constants';
+import type { CspAppContext } from '../../plugin';
+import type { CspRouter } from '../../types';
+import type { CspSetupStatus, Status } from '../../../common/types';
+import {
+  addRunningAgentToAgentPolicy,
+  getCspAgentPolicies,
+  getCspPackagePolicies,
+} from '../../lib/fleet_util';
+import { isLatestFindingsIndexExists } from '../../lib/is_latest_findings_index_exists';
+
+export const INDEX_TIMEOUT_IN_MINUTES = 10;
+
+const getHealthyAgents = (enrichedAgentPolicies: GetAgentPoliciesResponseItem[]): number =>
+  enrichedAgentPolicies.reduce(
+    (previousValue, currentValue) => previousValue + (currentValue.agents || 0),
+    0
+  );
+
+const getMinutesPassedSinceMoment = (momentInput: MomentInput): number =>
+  moment().diff(moment(momentInput), 'minutes');
+
+const getStatus = (
+  findingsIndexExists: boolean,
+  installedIntegrations: number,
+  healthyAgents: number,
+  minutesPassedSinceInstallation: number
+): Status => {
+  if (findingsIndexExists) return 'indexed';
+  if (installedIntegrations === 0) return 'not-installed';
+  if (healthyAgents === 0) return 'not-deployed';
+  if (minutesPassedSinceInstallation <= INDEX_TIMEOUT_IN_MINUTES) return 'indexing';
+  if (minutesPassedSinceInstallation > INDEX_TIMEOUT_IN_MINUTES) return 'index-timeout';
+
+  throw new Error('Could not determine csp setup status');
+};
+
+const getCspSetupStatus = async (
+  logger: CspAppContext['logger'],
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  packageService: PackageService,
+  packagePolicyService: PackagePolicyServiceInterface,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<CspSetupStatus> => {
+  const [findingsIndexExists, installationPackageInfo, latestPackageInfo, installedIntegrations] =
+    await Promise.all([
+      isLatestFindingsIndexExists(esClient, logger),
+      packageService.asInternalUser.getInstallation(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
+      packageService.asInternalUser.fetchFindLatestPackage(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
+      getCspPackagePolicies(soClient, packagePolicyService, CLOUD_SECURITY_POSTURE_PACKAGE_NAME, {
+        per_page: 10000,
+      }),
+    ]);
+
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    installedIntegrations.items,
+    agentPolicyService
+  );
+
+  const enrichedAgentPolicies = await addRunningAgentToAgentPolicy(
+    agentService,
+    agentPolicies || []
+  );
+  const healthyAgents = getHealthyAgents(enrichedAgentPolicies);
+  const installedIntegrationsTotal = installedIntegrations.total;
+  const latestPackageVersion = latestPackageInfo.version;
+
+  const status = getStatus(
+    findingsIndexExists,
+    installedIntegrationsTotal,
+    healthyAgents,
+    getMinutesPassedSinceMoment(installationPackageInfo?.install_started_at || 0)
```

Comment:
nit I've found an edge-case when using `install_started_at`
This field gives the time when the package was installed in fleet UI and NOT when a healthy agent was deployed using the integration.
The issue with that is when a user creates integration and it takes him more than the threshold to deploy it, the user will see index timeout immediately. 

What I suggest is to use the max time between cis package policy creation time and agent enrollment time and use that for calculation. When having multiple agents, I would take the minimum between all the healthy agents.

I'll discuss with product if it is critical for them, let's continue as it is now